core/commonMain/src/kotlinx/io/Input.kt

```diff
@@ -114,100 +114,61 @@ public abstract class Input : Closeable {
      *
      * @throws EOFException if no more bytes can be read.
      */
-    public fun readByte(): Byte {
-        val offset = position
-
-        if (limit > offset) {
-            position = offset + 1
-            return buffer.loadByteAt(offset)
-        }
-
-        if (fetchBuffer() == 0) {
-            throw EOFException("End of file while reading buffer")
-        }
-
-        position = 1
-        return buffer.loadByteAt(0)
-    }
-
-    /**
-     * Reads an [UByte] from this Input.
-     *
-     * @throws EOFException if no more bytes can be read.
-     */
-    public fun readUByte(): UByte = readByte().toUByte()
+    public fun readByte(): Byte = readPrimitive(
+            1,
+            { buffer, offset -> buffer.loadByteAt(offset) },
+            { it.toByte() })
 
     /**
      * Reads a [Long] from this Input.
      *
      * @throws EOFException if no more bytes can be read.
      */
     public fun readLong(): Long = readPrimitive(
-        8, { buffer, offset -> buffer.loadLongAt(offset) }
-    ) { it }
+        8,
+        { buffer, offset -> buffer.loadLongAt(offset) },
+        { it })
 
-    /**
-     * Reads a [ULong] from this Input.
-     *
-     * @throws EOFException if no more bytes can be read.
-     */
-    public fun readULong(): ULong = readPrimitive(
-        8, { buffer, offset -> buffer.loadULongAt(offset) }
-    ) { it.toULong() }
 
     /**
      * Reads a [Double] from this Input.
      *
      * @throws EOFException if no more bytes can be read.
      */
     public fun readDouble(): Double = readPrimitive(
-        8, { buffer, offset -> buffer.loadDoubleAt(offset) }
-    ) { Double.fromBits(it) }
+        8, { buffer, offset -> buffer.loadDoubleAt(offset) },
+        { Double.fromBits(it) })
 
     /**
      * Reads an [Int] from this Input.
      *
      * @throws EOFException if no more bytes can be read.
      */
     public fun readInt(): Int = readPrimitive(
-        4, { buffer, offset -> buffer.loadIntAt(offset) }
-    ) { it.toInt() }
-
-    /**
-     * Reads an [UInt] from this Input.
-     *
-     * @throws EOFException if no more bytes can be read.
-     */
-    public fun readUInt(): UInt = readPrimitive(
-        4, { buffer, offset -> buffer.loadUIntAt(offset) }
-    ) { it.toUInt() }
+        4,
+        { buffer, offset -> buffer.loadIntAt(offset) },
+        { it.toInt() })
```

Comment:
Codestyle

Reply:
I've left each lambda on its own line deliberately to improve readability.